File: hw/vizcaino/u2/hw14MongoDb/hw14MongoDBPy/dao/reservation_dao.py
from database.mongo_connection import MongoConnection

class ReservationDAO:

    def __init__(self):
        db = MongoConnection.get_database()
        self.collection = db["reservations"]

    def save(self, reservation):

        self.collection.insert_one({
            "id": reservation.id,
            "customerName": reservation.customer_name,
            "phone": reservation.phone,
            "status": reservation.status
        })

    def find_all(self):

        return list(self.collection.find({}, {"_id": 0}))

    def delete(self, id):

        self.collection.delete_one({"id": id})

File: hw/vizcaino/u2/hw14MongoDb/hw14MongoDBJS/dao/ReservationDAO.js
const { getDatabase } =
require("../config/mongoConnection");

class ReservationDAO {

    async save(reservation) {

        const db =
        await getDatabase();

        await db
        .collection("reservations")
        .insertOne(reservation);
    }

    async findAll() {

        const db =
        await getDatabase();

        return await db
        .collection("reservations")
        .find()
        .toArray();
    }

    async delete(id) {

        const db =
        await getDatabase();

        await db
        .collection("reservations")
        .deleteOne({ id });
    }
}

module.exports =
new ReservationDAO();